### PR TITLE
sunxi: add support for Libre Computer ALL-H3-CC H5

### PR DESCRIPTION
**Pull request by damex**
Specification:

- CPU: Allwinner H5, Quad-core Cortex-A53 Up to 1GHz
- DDR3 RAM: 2GB
- Network:
    10/100M Ethernet x 1
- IR: x1 (Receive)
- USB (Host) Type-A x3
- USB (OTG) Type-A x1
- MicroSD Slot x 1
- eMMC Slot x1
- MicroUSB power input
- GPIO 40pin header
- UART 3pin header
- Leds:
    - librecomputer:blue:status
    - librecomputer:green:pwr
- Buttons:
    - uboot button (used to enter fel mode)
    - power button (can trigger power on)
- Power Supply via MicroUSB or GPIO 5V/2A

Installation:

- Write the image to SD Card with dd
- Boot from the SD Card

Signed-off-by: Roman Kuzmitskii damex.pp@icloud.com

### CODE REVIEW

**Comment by damex** [comment]
are we okay with that kind of naming?
officially board have code name 'Tritium' and made by Libre Computer.
in kernel and uboot sources it is available as 'all-h3-cc' and have 'libretech' prefix.

more details about board could be found here:
https://linux-sunxi.org/Libre_Computer_Board_ALL-H3-CC
https://libre.computer/products/boards/all-h3-cc/

File: target/linux/sunxi/image/cortexa53.mk [hunk 1]
```diff
@@ -38,6 +38,14 @@ define Device/friendlyarm_nanopi-neo2
 endef
 TARGET_DEVICES += friendlyarm_nanopi-neo2
 
+define Device/librecomputer_tritium-h5
+  DEVICE_VENDOR := Libre Computer
+  DEVICE_MODEL := Tritium (H5)
```
Comment (adschm) [inline target/linux/sunxi/image/cortexa53.mk:43]: So, is "H5" an alternate name, part of the name, or a revision? In the first case it shouldn't be part of the device string/compatible, in the second and third case it shouldn't be put into brackets here ...

Comment (damex) [reply target/linux/sunxi/image/cortexa53.mk:43]: `Tritium` the name of the device.
`H5` is the part of the device variant (could i call that <board version>? it is released at the same time but have other pin compatible SoC).

there is three versions and all of them are called 'Tritium':
 - one with `H2+` SoC (arm32)
 - one with `H3` SoC (arm32)
 - one with `H5` SoC (arm64)

maybe we use `DEVICE_VARIANT` for that?
there has to be a way to distinguish between that three boards in a tree.
i would be glad to introduce other boards too.

**Comment by adschm** [comment]
> are we okay with that kind of naming?
> officially board have code name 'Tritium' and made by Libre Computer.
> in kernel and uboot sources it is available as 'all-h3-cc' and have 'libretech' prefix.

From what I remember, we follow the kernel vendor prefixes if they have established one.

In [1] we find:

libretech	Shenzhen Libre Technology Co., Ltd

[1] https://www.kernel.org/doc/Documentation/devicetree/bindings/vendor-prefixes.txt

**Comment by damex** [comment]
> > are we okay with that kind of naming?
> > officially board have code name 'Tritium' and made by Libre Computer.
> > in kernel and uboot sources it is available as 'all-h3-cc' and have 'libretech' prefix.
> 
> From what I remember, we follow the kernel vendor prefixes if they have established one.
> 
> In [1] we find:
> 
> libretech Shenzhen Libre Technology Co., Ltd
> 
> [1] https://www.kernel.org/doc/Documentation/devicetree/bindings/vendor-prefixes.txt


i did the adjustments accordingly to u-boot and kernel naming schemes.
although i used `Libre Computer` as `DEVICE_VENDOR` and `ALL-H3-CC H5` as `DEVICE_MODEL` since it is the same way that upstream kernel describe board.

after booting luci gets that string from upstream devicetree and call it 'Libre Computer Board ALL-H3-CC H5'

https://github.com/torvalds/linux/blob/master/arch/arm64/boot/dts/allwinner/sun50i-h5-libretech-all-h3-cc.dts#L11-L12

please check it again. i think that naming matches what upstream have.

**Comment by adschm** [comment]
I think the new naming scheme is better. One might put the H5 in DEVICE_VARIANT, but I don't really have a preference here so we might as well keep it.

**Comment by damex** [comment]
> I think the new naming scheme is better. One might put the H5 in DEVICE_VARIANT, but I don't really have a preference here so we might as well keep it.

sure, i just put H5 in DEVICE_VARIANT.

**Comment by damex** [comment]
could we merge it if everything is good?

**Comment by ynezz** [comment]
Thanks! Pulled into my staging tree at https://git.openwrt.org/openwrt/staging/ynezz.git

**Comment by predators46** [comment]
Hy @damex 

is there any plans to add Libre Computer Le Potato (amlogic s905x) i need this so i can use openwrt

**Comment by damex** [comment]
> Hy @damex
> 
> is there any plans to add Libre Computer Le Potato (amlogic s905x) i need this so i can use openwrt

Hi, sorry but i can not help you with that since i don’t have access for that board anymore.

**Comment by predators46** [comment]
> > Hy @damex
> > is there any plans to add Libre Computer Le Potato (amlogic s905x) i need this so i can use openwrt
> 
> Hi, sorry but i can not help you with that since i don’t have access for that board anymore.

😭😭😭

**Pull request closed without merging**